scraper/PageScraper.ts

```diff
@@ -0,0 +1,340 @@
+        coursesDataElements.push({
+          course_info: course_info,
+          classes: classes,
+        })
+      } else {
+        coursesDataElements.push({ course_info: course_info })
+      }
+    }
+    return coursesDataElements
+  })
+  return chunks
+}
+
+/**
+ * @constant { TermFinderReference }: Default reference to follow to find the correct term
+ */
+const defaultReference: TermFinderReference = [
+  { term: Term.Summer, census: '1/8' },
+  { term: Term.T1, census: '3/17' },
+  { term: Term.T2, census: '6/30' },
+  { term: Term.T3, census: '10/13' },
+  { term: Term.S1, census: '3/31' },
+  { term: Term.S2, census: '8/18' },
+]
+
+interface TermFinderParams {
+  course: Course
+  reference?: TermFinderReference
+}
+
+/**
+ * Given some reference dates for the term,
+ * the function returns one of the 6 terms that
+ * the course can be a part of
+ * [ Summer, T1, T2, T3, S1, S2 ]
+ * @param { Course } course: course the term is required for
+ * @param { TermFinderReference } reference The reference is optional. The default is:
+ * (Convert the dates into start (month of starting)
+ * and length (number of months the course runs))
+ * Summer: 26/11 ---> 10/02
+ * T1: 18/02 ---> 19/05
+ * T2: 03/06 ---> 01/09
+ * T3: 16/09 ---> 15/12
+ * S1: 25/02 ---> 30/06
+ * S2: 15/07 ---> 10/11
+ * The date format for reference census dates is month/day
+ * @returns { Term[] }: List of all the terms the course runs in based on the census dates provided
+ */
+const termFinder = ({
+  course,
+  reference = defaultReference,
+}: TermFinderParams): Term[] => {
+  if (!course.censusDates || course.censusDates.length <= 0) {
+    throw new Error('no census dates for course: ' + course.courseCode)
+  }
+  // For each of the census dates...add a term to the list
+  const censusDates = formatDates(course.censusDates)
+  const currentYear = new Date().getFullYear()
+
+  // Add the current year to the date and convert it to a date
+  // object so it can be easily parsed
+  const refCensusDates = formatDates(
+    reference.map(element => {
+      return element.census + '/' + currentYear
+    })
+  )
+
+  const termList: Term[] = []
+  for (const censusDate of censusDates) {
+    // minimum difference with the reference census date means the course is in that term
+    let min = -1
+    let index = 0
+    let term: Term
+    for (const refDate of refCensusDates) {
+      // Calculate the difference (in days) between the
+      // census date for the term and the reference census date
+      const diff = Math.abs(
+        Math.round(
+          (censusDate.getMonth() - refDate.getMonth()) * 30 +
+            censusDate.getDate() -
+            refDate.getDate()
+        )
+      )
+      if (diff < min || min === -1) {
+        min = diff
+        term = reference[index].term
+      }
+      index++
+    }
+    termList.push(term)
+  }
+  return termList
+}
+
+/**
+ * Function scrapes all the course data on the given page
```

Comment:
```suggestion
 * Scrapes all the course data on the given page
```

`Function` is redundant as we know it's a function

Reply:
Sure